src/mscorlib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -807,10 +807,6 @@ internal static String GetRealFormat(String format, DateTimeFormatInfo dtfi)
                 case 'M':       // Month/Day Date
                     realFormat = dtfi.MonthDayPattern;
                     break;
-                case 'o':
-                case 'O':
-                    realFormat = RoundtripFormat;
-                    break;
```

Comment:
Actually this is a good catch. the parsing code use this code path too. through  ExpandPredefinedFormat which called from DoStrictParse